src/quacc/recipes/torchsim/core.py

```diff
@@ -0,0 +1,318 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Literal, TypedDict
+
+import torch_sim as ts
+from torch_sim.autobatching import BinningAutoBatcher, InFlightAutoBatcher
+from torch_sim.models.interface import ModelInterface
+
+from quacc import job
+
+if TYPE_CHECKING:
+    import pathlib
+    from collections.abc import Callable
+
+    import torch
+    from ase.atoms import Atoms
+    from torch_sim.integrators import Integrator
+    from torch_sim.optimizers import Optimizer
+    from torch_sim.trajectory import TrajectoryReporter
+    from torch_sim.typing import StateLike
+
+    class TorchSimSchema(TypedDict):
+        atoms: list[Atoms]
+        model: str  # string of the model name
+
+    class TrajectoryReporterDetails(TypedDict):
+        state_frequency: int
+        trajectory_kwargs: dict[str, Any]
+        prop_calculators: dict[int, list[str]]
+        state_kwargs: dict[str, Any]
+        metadata: dict[str, str] | None
+        filenames: list[str | pathlib.Path] | None
+
+    class AutobatcherDetails(TypedDict):
+        autobatcher: Literal["BinningAutoBatcher", "InFlightAutoBatcher"]
+        memory_scales_with: Literal["n_atoms", "n_atoms_x_density"]
+        max_memory_scaler: float | None
+        max_atoms_to_try: int | None
+        memory_scaling_factor: float | None
+        max_iterations: int | None
+        max_memory_padding: float | None
+
+    class TorchSimOptSchema(TorchSimSchema):
+        optimizer: Optimizer
+        max_steps: int
+        steps_between_swaps: int
+        trajectory_reporter: TrajectoryReporterDetails | None
+        autobatcher: AutobatcherDetails
+        init_kwargs: dict[str, Any] | None
+        optimizer_kwargs: dict[str, Any]
+
+    class TorchSimIntegrateSchema(TorchSimSchema):
+        integrator: Integrator
+
+        n_steps: int
+        temperature: float | list[float]
+        timestep: float
+        trajectory_reporter: TrajectoryReporterDetails | None
+        autobatcher: AutobatcherDetails
+        integrator_kwargs: dict[str, Any]
+
+    class TorchSimStaticSchema(TorchSimSchema):
+        all_properties: list[dict[str, torch.Tensor]]
+        trajectory_reporter: TrajectoryReporterDetails | None
+        autobatcher: AutobatcherDetails
```

Comment:
If you think these type hints might get used in several places, they can go in `quacc.types`.

Reply:
If we rewrite the MLIP interface to use TorchSim that might be necessary but otherwise I'd hold off to keep the import localized.